plugins/woocommerce/templates/single-product/add-to-cart/variable.php

```diff
@@ -45,13 +45,20 @@
 										'product'   => $product,
 									)
 								);
-								echo end( $attribute_keys ) === $attribute_name ? wp_kses_post( apply_filters( 'woocommerce_reset_variations_link', '<a class="reset_variations" href="#">' . esc_html__( 'Clear', 'woocommerce' ) . '</a>' ) ) : '';
 							?>
 						</td>
 					</tr>
 				<?php endforeach; ?>
+				<tr>
+					<td colspan="2">
+						<button class="reset_variations wp-element-button" aria-label="<?php echo esc_html__( 'Clear options', 'woocommerce' ); ?>">
```

Comment:
Looks like previously this was an anchor link and we're switching that to a button (and a slight change to how this is output). The concern I have here is it seems we're getting rid of the existing `woocommerce_reset_variations_link` filter, is that correct? Wondering if we can keep that or if not, we'll probably need to deprecate that for anyone that may be utilizing it.

Reply:
@dkotter I removed it as it felt like we just needed to have that button outside the actual variation options to make it easier for screen readers to navigate. If you think that there is a need to add it back in, maybe I am not understanding its purpose?

Reply:
@amesplant the issue is the removing of the filter. That's a breaking change for anyone using the filter. As mentioned by @dkotter if we want to remove a filter we really need to deprecate it to give chance to find an alternate solution.

Reply:
@dkotter cc @samueljseay totally understand and happy to reinstate the filter, I will get that done today.

I am also working on the tests as well as the css since they are looking for a link and not a button.

Reply:
@dkotter cc @samueljseay `woocommerce_reset_variations_link` has been reinstated :D, and tests and css have been addressed.

Do we need to add a test for the `reset` button since additional functionality has been included? In addition to its normal behavior, I have added a screen reader announcement for when the button has been selected.